### ISSUE TITLE
Hash scrolling rubber-bands when content changes

### ISSUE THREAD

**Issue by denis-sokolov** [nozzle/react-static#449]
Scenario:

1. Bet a URL `/example#foo`, scrolled down to the end of the page.
2. Click on a link `#bar`.
3. `onhashchange` a piece of custom code removes `<div id="foo">`, adds a `<div id="bar">` in its place.
4. Screen shows the new div, and the page instantly scrolls to the top of `<div id="bar">`.
5. With no delay start scrolling down.

Expected:

6. Page scrolls down.

Actual:

6. Page scrolls down a small amount and is immediately scrolled to the top of `<div id="bar">`. This repeats for a better part of a second, after which the scroll is released to act naturally.

_I’m sorry to bring the hash scrolling again, you must hate it by now._

**Comment by tannerlinsley** [comment]
Hmmm, If possible, could you spin up a repo with this?

**Comment by JustFly1984** [comment]
I disabled react-static scrolling:
```
<Router
          history={history}
          autoScrollToTop={false}
        >
          <Routes />
</Router>
```

This pattern for scroll to top:

```
    componentDidMount () {
      if (typeof document !== 'undefined') {
        window.scrollTo(0, 0)
      }

```

and this pattern instead of #scroll

```
getRef = div => {
    this.div = div
}

componentDidMount () {
    this.div.scrollIntoView({
        block: 'start',
        behavior: 'smooth'
    })
}

render () {
 <div ref={this.getRef} />
}
```

**Comment by denis-sokolov** [comment]
Couldn’t come up with a demo, came up with two fixes #460, #462.
